### PR TITLE
fix: logits processors for CB 

### PR DESCRIPTION
**Pull request by wallashss**
# Description

This PR introduces a wrapper of the logits processor that are injected to vlllm that will handle the logits processor in a distributed way. The wrapper is initialized with the logits class and the batch_size. So, from the logits processor perspective it will "think" that it's only handling a request per step, while the wrapper receives the batch of logits, slice and redistribute for each request.

### CODE REVIEW

**Comment by wallashss** [comment]
bot:test

File: tests/e2e/test_sampling_params.py [hunk 1]
```diff
@@ -7,18 +7,19 @@
 
 pytestmark = [pytest.mark.full_model, pytest.mark.other_e2e]
 
+# TODO: REVERT THIS CHANGE!
```
[the remark this answers is not in the document]
Comment (yannicks1) [reply tests/e2e/test_sampling_params.py]: or do we already have a similar test for CB?

[the remark this answers is not in the document]
Comment (wallashss) [reply tests/e2e/test_sampling_params.py]: We do not have this test for CB. I think the issue here is increase too much the time of CI. Moreover, they do not repro very well the issue of this PR.

**Comment by wallashss** [comment]
bot:test

**Comment by wallashss** [comment]
bot:test
